### ISSUE TITLE
Add support for Asio4All

### ISSUE THREAD

**Issue by PeterHevesi** [FluidSynth/fluidsynth#1367]
Would you please add support for Asio4All? 
That would be super awesome :)
Thanks so much :)

**Comment by derselbst** [comment]
I don't think this will happen. I am the only active maintainer around here. There is quite little overall activity currently due to time constraints. And there are a couple of topics I consider to be more important. Also, I don't have any ASIO devices that I could use for testing. A PR would be welcome, though I consider this to be unlikely.

**Comment by PeterHevesi** [comment]
Sure, I understand :)
Actually, I didn't expect that you will do this, but I tried, because you never know :D

We are developing music game in Unity and we used one plugin called MPTK Maestro, which uses your FluidSynth.
It is very good, but when we wanted to directly play sounds with it that reacts to user input through keyboard, it had high latency, so we can't use it for some of our goals we want to achieve.

We know, that through ASIO supported sound cards, we can achieve very low latency, and we own such sound cards, but the problem is, that players of our game often won't have these sound cards, so we were exploring ways to achieve low latency even without it.
Just through the standard internal sound card on the motherboard. 

And it is indeed possible through Asio4All, which is some library or driver that can achieve fantastic low latency, without owning any asio device :)
So you don't need them for Asio4All, only when using standard asio driver. Asio4All just runs on every windows PC. 

Meanwhile, we already found 1 asset on Unity Store, that allows us to use this Asio4All directly in Unity and we could achieve perfect latency, but is is very simple and have problems when playing multiple sounds at the same time.

So that's why I am interested in this, to be able to have this fantastic latency Asio4All provides, but at the same time having all the other benefits and functionality of a system like your FluidSynth.

But if you are the only one that develops this (which I didn't know) , I absolutely understand you won't prioritize this. 

But anyway, thank you for your fast reply :)

**Comment by jimhen3ry** [comment]
If you are concerned about latency on Windows, making use of the WASAPI interface rather than DirectIO might be helpful.
